stl/inc/span

```diff
@@ -0,0 +1,679 @@
+// span standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SPAN_
+#define _SPAN_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if _HAS_CXX20
+#include <array>
+#include <cstddef>
+#include <type_traits>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+inline constexpr size_t dynamic_extent = static_cast<size_t>(-1);
+
+// STRUCT TEMPLATE _Span_iterator
+template <class _Ty>
+struct _Span_iterator {
+#ifdef __cpp_lib_concepts
+    using iterator_concept = contiguous_iterator_tag;
+#endif // __cpp_lib_concepts
+    using iterator_category = random_access_iterator_tag;
+    using value_type        = remove_cv_t<_Ty>;
+    using difference_type   = ptrdiff_t;
+    using pointer           = _Ty*;
+    using reference         = _Ty&;
+
+    constexpr _Span_iterator() noexcept = default;
+
+#if _ITERATOR_DEBUG_LEVEL >= 1

```

Comment:
@StephanTLavavej @CaseyCarter @barcharcraz Do we want to only do this for IDL2 in new code given that IDL1's days are numbered?

Reply:
I'm happy enough to be consistent with other iterators that I wouldn't suggest a change, but I wouldn't argue if someone made the change either.

Reply:
Since there are essentially no IDL1 users, I am favor of global consistency.

Reply:
Sounds good then, no change requested.